### PR TITLE
[Jobs Table: 1 of 2] Add additional structs for Jobs table

### PR DESCRIPTION
**Pull request by jfurrow**
These structs will be later used by the Jobs table.

This PR introduces two methods on the `Jobs` struct:
* `#getLastRunStatus` retrieves the status and time of the most recent run
* `#getScheduleStatus` retrieves the status of the schedule

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @kennyt

File: src/js/structs/Job.js [hunk 1]
```diff
@@ -18,11 +19,42 @@ module.exports = class Job extends Item {
     return this.get('id');
   }
 
+  getLastRunStatus() {
+    let {lastFailureAt, lastSuccessAt} = this.get('status');
```
Comment (pierlo-upitup) [inline src/js/structs/Job.js]: Let's also account for undefined statuses (ie a one-off job that has never run).

File: src/js/structs/Job.js [hunk 2]
```diff
@@ -18,11 +19,42 @@ module.exports = class Job extends Item {
...
+    return {status: 'Success', time: lastSuccessAt};
+  }
+
   getName() {
     return this.getId().split('.').pop();
   }
 
   getSchedules() {
-    return this.get('schedules');
+    return this.get('schedules') || [];
```
Comment (orlandohohmeier) [inline src/js/structs/Job.js]: Not in this PR burt can we please also introduce a schedule struct?

**Comment by MatApple** [comment]
Flicts

File: src/js/structs/Job.js (hunk 1 above)
Comment (jfurrow) [inline src/js/structs/Job.js]: Done!

File: src/js/structs/Job.js (hunk 2 above)
Comment (jfurrow) [inline src/js/structs/Job.js]: Makes sense! JIRA here: https://mesosphere.atlassian.net/browse/DCOS-7691

**Comment by jfurrow** [comment]
Conflicts & feedback addressed!

File: src/js/structs/Job.js [hunk 3]
```diff
@@ -32,11 +33,56 @@ module.exports = class Job extends Item {
     return mem;
   }
 
+  getLastRunStatus() {
+    let {lastFailureAt = null, lastSuccessAt = null} = this.get('status');
```
Comment (kennyt) [inline src/js/structs/Job.js]: can we make a getter for `get('status')`?

**Comment by kennyt** [comment]
after my one comment + tests for it, i'll merge

File: src/js/structs/Job.js (hunk 3 above)
Comment (jfurrow) [inline src/js/structs/Job.js]: Done!

**Pull request merged**